src/components/FileUpload.tsx

```diff
@@ -54,16 +55,105 @@ export default function FileUpload() {
         setErrorOpen(false);
     };
 
+    async function getToken() {
+        const token = await firebase.auth().currentUser?.getIdToken();
+
+        if (token === undefined) return {};
```

Comment:
```suggestion
        if (token === undefined) return "";
```
I think this will fix your type errors where you had to do `as unknown as string`